### DISCUSSION TITLE
Example of asking a dynamic confirm question

### DISCUSSION THREAD

**Discussion by simplenotezy** [jmcdo29/nest-commander#823] [Q&A] [answered]
I'd like to ask a dynamic question (by using the InquirerService directly) that I'd like the user to confirm, but can't find an example of this.

**Comment by simplenotezy** [comment, marked as answer]
```ts
      await inquirer.inquirer.prompt({
        name: 'confirm',
        type: 'confirm',
        message: "Please confirm that you'd like to XXX",
      });

**Comment by stefaanMLB** [comment]
Can a dynamic question be inserted into a question sequence defined with the decorator syntax ?

**Reply by jmcdo29** [reply]
I'm not quite sure I understand the question here

**Reply by stefaanMLB** [reply]
The question sequences and possible answers/choices are defined statically in classes with `@QuestionSet()` decorator.
Is it possible to make the choices (of f.i. a list question) depend on the state of a nest provider ?

**Reply by jmcdo29** [reply]
[Actually, yes](https://github.com/jmcdo29/unteris/blob/7732ebc94f23b5f4a26afd1558676b5da1bb8aa0/apps/cli/src/app/seeds/deity.questions.ts#L75), that's what the [`@ChoicesFor()`](https://nest-commander.jaymcdoniel.dev/en/api/#inquirer-decorators) decorator is there for, so that you can have the [choices (or validation or whatever)](https://nest-commander.jaymcdoniel.dev/en/features/inquirer/#inquirer-decorators) depend on something from the provider's state context in a dynamic way.

**Reply by stefaanMLB** [reply]
Ok, yes that is what I'm looking for.
Thanks for the help !
